include/service/entities/site_service.js

```diff
@@ -123,6 +142,16 @@ module.exports = function SiteServiceModule(pb) {
     };
 
     /**
+     * Get all of the redirect_hosts in the database
+     * @method getRedirectHosts
+     * @param {Function} cb - the callback function
+     */
+    SiteService.prototype.getRedirectHosts = function(cb) {
+      var dao = new pb.DAO();
+      dao.q(REDIRECT_HOSTS, { select: pb.DAO.SELECT_ALL, where: {} }, cb);
+    };
```

Comment:
I think you would want to store all of the previous hostnames in an array or hash inside of the site object itself.  That way the site itself is aware of its evolution over time.